### PR TITLE
Add spackage for py-sphinx-rtd-dark-mode

### PR DESCRIPTION
**Pull request by ktsai7**
Add spackage for py-sphinx-rtd-dark-mode

The package belongs to [MrDogeBro](https://github.com/MrDogeBro/sphinx_rtd_dark_mode/commits?author=MrDogeBro)

### CODE REVIEW

**Review by tldahlgren — COMMENTED** [review]
Confirmed the sh256.

**Comment by tldahlgren** [comment]
Added python gurus as reviewers.

File: var/spack/repos/builtin/packages/py-sphinx-rtd-dark-mode/package.py [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+class PySphinxRtdDarkMode(PythonPackage):
+    """A Sphinx extension for adding a toggleable dark mode to the Read the Docs theme."""
+
+    homepage = "https://github.com/MrDogeBro/sphinx_rtd_dark_mode"
+    pypi = "sphinx-rtd-dark-mode/sphinx_rtd_dark_mode-1.2.4.tar.gz"
+
+    version("1.2.4", sha256="935bc1f3e62fc76eadd7d2760ac7f48bab907a97e44beda749a48a2706aeed63")
+
+    depends_on("py-setuptools", type="build")
+    depends_on("py-sphinx", type=("build", "run"))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-sphinx-rtd-dark-mode/package.py]: sphinx doesn't seem to be imported in this package

Comment (ktsai7) [reply var/spack/repos/builtin/packages/py-sphinx-rtd-dark-mode/package.py]: `depends_on("py-sphinx", type=("build", "run"))` is not sufficient to pull in sphinx?

Comment (adamjstewart) [reply var/spack/repos/builtin/packages/py-sphinx-rtd-dark-mode/package.py]: I'm saying I don't think Sphinx is a dependency

Comment (ktsai7) [reply var/spack/repos/builtin/packages/py-sphinx-rtd-dark-mode/package.py]: ahh, got it.

**Review by adamjstewart — APPROVED** [review]

**Pull request merged**